### PR TITLE
Move pydrake.solvers.ik => pydrake.attic.solvers.ik

### PR DESCRIPTION
**Pull request by EricCousineau-TRI**
Relates #10207 

@jwnimmer-tri I think that having the `attic/` for `pydrake` is super useful (and I guess part of your plan in #9783 and #9314?)

Here're some of my initial thoughts on the Python side:
- If we transition code, it should immediately be public. Trying to mess with symlinks or anything of that sort just sounds messy and painful.
- It is super useful to move stuff to the attic early, as having generic names like `collision` and `joints` as RBT modules next to `parsing` and `tree` MBP modules is super confusing (and I guess the motivator for `attic` in the first place?)

Would it be OK with you if we possibly expose `pydrake.attic` before we expose `drake/attic/`?
Or did you have another shim plan in mind?

(Putting these thoughts here because I haven't seen an explicit mention of it yet in any issues, searching for `pydrake attic`.)

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/10255)

### CODE REVIEW

**Comment by jwnimmer-tri** [comment]
> @jwnimmer-tri I think that having the attic/ for pydrake is super useful (and I guess part of your plan in #9783 and #9314?)

Actually, I don't think I've proposed (nor even considered) having `attic` be a module name in pydrake.

> Would it be OK with you if we possibly expose pydrake.attic before we expose drake/attic/?  Or did you have another shim plan in mind?

I do not intend to expose `drake/attic` as a C++ include path.  The attic is _only_ a git organizational tool (that therefore also leaks into Bazel package names, but no further).

That said, it still _might_ make sense for pydrake to add an explicit `attic`, if there is a good reason for it.

> ...having generic names like collision and joints as RBT modules next to parsing and tree MBP modules is super confusing (and I guess the motivator for attic in the first place?)

Sure, I think it's fair to consider what the pydrake docs look like.  However, my proposed solution would be to exclude _all_ attic items from the generated documentation.  They would still have live `help()` strings to e.g. explain their arguments, but really there's no need to have those docstrings on the public website, is there?  We're only keeping the code around so _existing_ users can keep using their _existing_ code.  They lived without website docstrings for 4 years, I don't think losing them after two months is a big deal.

There may be cause to rework the _build_ system for attic'd code a bit, e.g., to reduce false coupling between attic and non-attic code.  My intuition says, though, that we can do that without introducing a new public module name.

In short: I am not yet convinced that this is a good direction to move in.

**Comment by EricCousineau-TRI** [comment]
> That said, it still might make sense for pydrake to add an explicit attic, if there is a good reason for it.

If we're OK with it not being documented, then I'm generally OK with awkward shims 'n such, and moving stuff to `pydrake._attic` to simplify development.

However, if it will be long-lived (say more than 3 months?), I'd like to avoid something like this happening:
* New user downloads some kewl code, that code imported enough modules to pull in both MBP and RBT bits.
* User explores using tab-complete in Jupyter. Comes across `pydrake.multibody.plant` and `pydrake.multibody.collision`, and then proceeds to waste 5min exploring methods via tab-complete to see how they combine the two; *if* they look at `help()` or the website, they'll realize that one of these things is not like the other... Yuck.

I'd be OK with that happening for another 2 months, but really don't want to leave that in the open for too long.

I guess my question boils down to: How long do we intend to keep the `attic/` accessible at all (via `drake/{path}`, or `drake/attic/{path}`)?
